### DISCUSSION TITLE
SKMetalView on iOS.

### DISCUSSION THREAD

**Discussion by janne-hmp** [mono/SkiaSharp#2944] [General]
Since SKGLView still gets often corrupted on iOS in 3.0 Preview 3.1 (see #2840) in .NET MAUI (Xamarin works fine with 2.88.8, and there are no GL handlers in 2.88.8 for .NET MAUI), I was trying to make another control that uses SKMetalView on iOS instead as a PlatformView (rather than iOS.SKGLView). However, it looks like SKMetalView's DLL is not included on the iOS build (app crashes at start with an error related to missing dependency).

@mattleibow Would it be possible to include SKMetalView DLL also on iOS in SkiaSharp (now possibly deleted by a trimmer or the like)? At the same time, SKTouchHandler should be made public, so that a self-made control can access the class.

Even better, could we alternatively get a variable to SKGLView through which you can select the implementation (e.g., bool UsePlatformSpecificRendering), which does nothing on non-iOS devices but on iOS changes the implementation to SKMetalView? Or another control using SKMetalView on iOS and the the same rendering as SKGLView on other platforms, if the variable is difficult?

**Comment by taublast** [comment]
Can't tell about the above issue related to .NET 9, but in net8 it works fine for me with

` <PackageReference Include="SkiaSharp.Views.Maui.Controls.Compatibility" Version="2.88.9-preview.2.2" />`

with

`builder.UseSkiaSharp(true);'

this should work fine on iOS only, not simulator, not catalyst. But, the next one might work fine in catalyst, i don't remember about the simulator, normally should work too:

`<PackageReference Include="SkiaSharp.Views.Maui.Controls" Version="3.0.0-preview.5.4" />`

and

`builder.UseSkiaSharp();`
